packages/core/integration-tests/test/webmanifest.js

```diff
@@ -0,0 +1,154 @@
+
+    await assert.rejects(
+      () =>
+        bundle(
+          path.join(__dirname, '/integration/webmanifest-schema/index.html'),
+        ),
+      {
+        name: 'BuildError',
+        message: path.normalize('Invalid webmanifest'),
+        diagnostics: [
+          {
+            codeFrame: {
+              code: manifest,
+              codeHighlights: [
+                {
+                  end: {
+                    column: 5,
+                    line: 12,
+                  },
+                  message: 'Missing property src',
+                  start: {
+                    column: 5,
+                    line: 9,
+                  },
+                },
+                {
+                  end: {
+                    column: 6,
+                    line: 13,
+                  },
+                  message: 'Missing property src',
+                  start: {
+                    column: 5,
+                    line: 13,
+                  },
+                },
+                {
+                  end: {
+                    column: 19,
+                    line: 15,
+                  },
+                  message: 'Expected type array',
+                  start: {
+                    column: 18,
+                    line: 15,
+                  },
+                },
+              ],
+            },
+            filePath: manifestPath,
+            language: 'json',
+            message: 'Invalid webmanifest',
+            origin: '@parcel/transformer-webmanifest',
+          },
+        ],
+      },
+    );
+  });
+
+  it('should throw on missing dependency', async function() {
+    let manifestPathRelative =
+      './integration/webmanifest-not-found/manifest.webmanifest';
+    let manifestPath = path.join(__dirname, manifestPathRelative);
+    let manifest = await inputFS.readFileSync(manifestPath, 'utf8');
+
+    let message = `Failed to resolve 'icon.png' from '${escapeMarkdown(
+      manifestPathRelative,
+    )}'`;
+
+    await assert.rejects(
+      () =>
+        bundle(
+          path.join(__dirname, '/integration/webmanifest-not-found/index.html'),
+        ),
+      {
+        name: 'BuildError',
+        message,
+        diagnostics: [
+          {
+            codeFrame: {
+              code: manifest,
+              codeHighlights: [
+                {
+                  end: {
+                    column: 23,
+                    line: 5,
+                  },
+                  start: {
+                    column: 14,
+                    line: 5,
+                  },
+                },
+              ],
+            },
+            message,
+            filePath: manifestPath,
+            origin: '@parcel/core',
+          },
+          {
+            hints: [],
+            message: `Cannot load file './icon.png' in '${path.dirname(
+              manifestPathRelative,
+            )}'.`,
+            // origin: '@parcel/resolver-default',
```

Comment:
The fix for that is
```patch
diff --git packages/core/core/src/requests/PathRequest.js packages/core/core/src/requests/PathRequest.js
index eacced75..3492e4b6 100644
--- packages/core/core/src/requests/PathRequest.js
+++ packages/core/core/src/requests/PathRequest.js
@@ -211,11 +211,11 @@ export class ResolverRunner {
           }

           if (result.diagnostics) {
-            if (Array.isArray(result.diagnostics)) {
-              diagnostics.push(...result.diagnostics);
-            } else {
-              diagnostics.push(result.diagnostics);
-            }
+            let errorDiagnostic = errorToDiagnostic(
+              new ThrowableDiagnostic({diagnostic: result.diagnostics}),
+              resolver.name,
+            );
+            diagnostics.push(...errorDiagnostic);
           }
         }
       } catch (e) {
```

After that, you should be able to uncomment the line.

Reply:
Sure! I usually refrain from modifying other files in pull requests but I guess this is needed.